### PR TITLE
[RD-19157] Coalesce DF on write

### PR DESCRIPTION
**Pull request by ramivaknin**

### CODE REVIEW

File: src/main/scala/MLlibDriver.scala [hunk 1]
```diff
@@ -212,7 +203,7 @@ object MLlibDriver {
       case _ => throw new IllegalArgumentException(s"Model $modelName is not supported")
     }
     is.close()
-    predictions.write.mode(SaveMode.Overwrite).json(predictionsPath)
+    predictions.coalesce(1).write.mode(SaveMode.Overwrite).json(predictionsPath)
```
Comment (michaelvashchinsky) [inline src/main/scala/MLlibDriver.scala:206]: Are we aware of the warning? (http://spark.apache.org/docs/latest/api/java/index.html?org/apache/spark/ml/regression/package-summary.html)
image

Comment (DorHanegbySB) [reply src/main/scala/MLlibDriver.scala:206]: Under the assumption of a small dataset, we shouldn't worry about it. I don't think the volumes of data we are handling are even near that.

can we have OOM issues? @meirmaor WDYT?

Comment (meirmaor) [reply src/main/scala/MLlibDriver.scala:206]: If it's only the writing part which happens on a single node this is a non issue.

Comment (meirmaor) [reply src/main/scala/MLlibDriver.scala:206]: Is this correct? Does this cause order preservation? seems unlikely?

Comment (ramivaknin) [reply src/main/scala/MLlibDriver.scala:206]: it reduces the number of partitions on before write to 1 so only a single output file is produced and the DP method which unite the files does actually nothing

**Pull request merged**